examples/capture/main.rs

```diff
@@ -108,14 +127,35 @@ async fn run() {
     }
 
     if let Ok(()) = buffer_future.await {
-        let data = output_buffer.get_mapped_range(0, wgt::BufferSize::WHOLE);
-        let mut png_encoder = png::Encoder::new(File::create("red.png").unwrap(), size, size);
+        let padded_buffer = output_buffer.get_mapped_range(0, wgt::BufferSize::WHOLE);
+        // There are two transforms occuring here:
+        // 1.   map bgra -> rgba
+        // 2.   remove padding from buffer
+        let mut real_buffer = vec!(0; width * height * bytes_per_pixel);
+        for x in 0..width as usize {
+            for y in 0..height as usize {
+                let padded_offset = x + y * padded_width as usize;
```

Comment:
let's make this in bytes, and compute as `x * 4 + y * padded_bytes_per_row` instead